src/Features/Core/Portable/CodeFixes/Suppression/ConfigureSeverityLevelCodeAction.cs

```diff
@@ -0,0 +1,284 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Suppression
+{
+    internal sealed class ConfigureSeverityLevelCodeAction : CodeAction.CodeActionWithNestedActions
+    {
+        public ConfigureSeverityLevelCodeAction(
+            Diagnostic diagnostic,
+            ImmutableArray<CodeAction> nestedActions)
+            : base(FeaturesResources.Configure_severity_level_via_editorconfig_file,
+            ImmutableArray<CodeAction>.CastUp(nestedActions), isInlinable: false)
+        {
+        }
+
+        internal static async Task<Solution> ConfigureEditorConfig(
+            string severity,
+            Diagnostic diagnostic,
+            Project project,
+            Dictionary<string, Option<CodeStyleOption<bool>>> languageSpecificOptions,
+            Dictionary<string, Option<CodeStyleOption<ExpressionBodyPreference>>> expressionOptions,
```

Comment:
this seems like a layering violation.  This is a generic system for configuring editorconfig.  Why does it have any knowledge of specific enums like ExpressionBodyPreference?  Would we expect that every time we added a new optoin, we'd need to come fix up this code to be aware of it?